### ISSUE TITLE
memory issue

### ISSUE THREAD

**Issue by graygrayh** [rs/SDWebImage#1934]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Xcode7.3 ios10.3 , when load GIF , memory will Constantly Rising until app crash

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by mythodeia** [comment]
v4 now uses FLAnimatedImage to support gif loading. check [this](https://github.com/rs/SDWebImage#animated-images-gif-support)

**Comment by graygrayh** [comment]
yes,I use FLAnimatedImage,but memory will Constantly Rising until app crash
last year, it's well.  after ios10.3 ,this issue is appear
 Xcode7.3
 ios10.3

**Comment by mythodeia** [comment]
whats the size a the gif files you are trying to open

**Comment by graygrayh** [comment]
the file size is 892k .  I promise you will find this issue ,when you build a project loading gif with Xcode 7.3 and run it on the platform whose version is ios 10.3 . I think it's a bug from apple , but i cant resolve it

**Comment by mythodeia** [comment]
can you replicate this in the demo project (using v4 of this library)?

**Comment by graygrayh** [comment]
i have replicate this in demo project (please use xcode 7.3 , ios 10 later , you will find it)

**Comment by graygrayh** [comment]
another, language version swift 2.3

**Comment by mythodeia** [comment]
does this happen in simulator only or device? if this happens only in iOS 10.3? in iOS 9 the issue is not there? weird...if its an Apple firmware bug i do not know if there is a way to resolve this. Have you opened an issue in the FLAnimatedImage repository?

**Comment by graygrayh** [comment]
there  is not the issue in iOS 9. this happen not only on FLAnimatedImage, but also SwiftyGif, YLGIFImage, UIImageView. I think i should upgrade Xcode version and swift version to avoid it. It will be a huge project --!
